pkg/ingress/controller/controller.go

```diff
@@ -909,15 +928,21 @@ func (c *Controller) ensureIngress(ing *nwv1.Ingress) error {
 	address := lb.VipAddress
 	// Allocate floating ip for loadbalancer vip if the external network is configured and the Ingress is not internal.
 	if !isInternal && c.config.Octavia.FloatingIPNetwork != "" {
-		logger.Info("creating floating IP")
 
-		description := fmt.Sprintf("Floating IP for Kubernetes ingress %s in namespace %s from cluster %s", ingName, ingNamespace, clusterName)
-		address, err = c.osClient.EnsureFloatingIP(false, lb.VipPortID, c.config.Octavia.FloatingIPNetwork, description)
+		floatingIPSetting := getStringFromIngressAnnotation(ing, IngressAnnotationFloatingIP, "")
 		if err != nil {
-			return fmt.Errorf("failed to create floating IP: %v", err)
+			return fmt.Errorf("unknown annotation %s: %v", IngressAnnotationFloatingIP, err)
 		}
 
-		logger.WithFields(log.Fields{"fip": address}).Info("floating IP created")
+		description := fmt.Sprintf("Floating IP for Kubernetes ingress %s in namespace %s from cluster %s", ingName, ingNamespace, clusterName)
+
+		if floatingIPSetting != "" {
+			logger.Info("try to use floating IP : ", floatingIPSetting)
```

Comment:
Missing `%s`? And unnecessary space before colon?

Reply:
For `%s` missing, nop seem to be ok:
<img width="517" alt="image" src="https://github.com/kubernetes/cloud-provider-openstack/assets/18497863/11e90f41-a799-4124-96d7-2fdc7f3a1d7d">

For space before colon, ok fixed :)

Ok for the weirdly EnsureFloatingIP() structure, I will try to improve it tomorrow.

Reply:
Should be better now, I let you check